src/Schema/Directives/CanDirective.php

```diff
@@ -95,7 +112,25 @@ function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo)
                 $gate = $this->gate->forUser($context->user());
                 $checkArguments = $this->buildCheckArguments($args);
 
-                foreach ($this->modelsToCheck($resolveInfo->argumentSet, $args) as $model) {
+                if ($this->directiveArgValue('findByArgs') === true) {
+                    if ($this->directiveArgValue('find') !== null) {
+                        throw new DefinitionException("Can not use the argument 'find' when 'findByArgs' is true");
+                    }
+                    $models = $resolveInfo
```

Comment:
Given this is resolving a list of models to check permissions against, this should live in `modelsToCheck()`.